Python/bytecodes.c

```diff
@@ -1158,31 +1169,43 @@ dummy_func(
             }
         }
 
-        inst(LOAD_NAME, ( -- v)) {
-            PyObject *name = GETITEM(frame->f_code->co_names, oparg);
-            PyObject *locals = LOCALS();
-            if (locals == NULL) {
+        op(_LOAD_NAME_INTRO, (-- mod_or_class_dict, name)) {
```

Comment:
A couple of issues here:
* Why push the name onto the stack in the first part of the instruction, rather than fetch from `names` in the second part?
* The ops should be named for what they do, not how or where they are used. I.e. rename `_LOAD_NAME_INTRO` to something like `LOAD_LOCALS_DICT`

I'd break these up as follows:

`op(_LOAD_FROM_DICT_OR_GLOBALS)` -- load name in `oparg` from dict TOS, falling back on the globals dict.
`inst(LOAD_FROM_DICT_OR_GLOBALS)  = _LOAD_FROM_DICT_OR_GLOBALS;`
`inst(LOAD_NAME) = LOAD_LOCALS_DICT + _LOAD_FROM_DICT_OR_GLOBALS;`




The name should describe what it does, not how it is (currently) used.
This pushes the locals dict onto the stack.
Maybe name it `LOAD_LOCALS_DICT`
(I think `PUSH_LOCALS_DICT` would be a more descriptive name, but we use `LOAD` not `PUSH` for other instructions)

Reply:
This doesn't actually push onto the stack, it just saves it in a local variable since that's how these micro-ops work. You can see the final code that is output in the `generated_cases.c.h` changes for this diff (collapsed by default, buit LOAD_CLASSDICT_OR_GLOBAL is at line 1718).

I can rename the instructions, though.

Reply:
The stack between micro-ops may be conceptual not physical, but it is still a stack.

Why load the name in the first op when it isn't used there, why not load it in the second op where is used?
That way, the first op is only responsible for push the dict, and the name is effectively a constant in the second op instead of a variable.

Reply:
Good point about the name. I think I did it this way just to preserve the order of some operations in the existing LOAD_NAME, but that shouldn't make a difference.

Reply:
Reorganized this as suggested.

I realized "dict" is pedantically inaccurate as it could be a non-dict mapping, but I think it's fine as a mnemonic.